controllers/operator/routes.go

```diff
@@ -277,13 +265,41 @@ func (r *AuthenticationReconciler) handleRoutes(ctx context.Context, instance *o
 	return
 }
 
+func (r *AuthenticationReconciler) removeIdauth(ctx context.Context, instance *operatorv1alpha1.Authentication) (err error) {
+	namespace := instance.Namespace
+	reqLogger := log.WithValues("func", "ReconcileRoute", "namespace", namespace)
+	reqLogger.Info("Determined platform-id-auth Route should not exist; removing if present")
+	observedRoute := &routev1.Route{}
+	err = r.Get(ctx, types.NamespacedName{Name: "platform-id-auth", Namespace: namespace}, observedRoute)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to get existing platform-id-auth route for reconciliation")
+		return 
+	}
+	err = r.Delete(ctx, observedRoute)
+	if err != nil {
```

Comment:
Given what you've seen in testing, I would use the same flow control stanza as L274-279 here as well - we talked before about the case where, if the Route had been deleted already by the time reaches the `r.Delete` call, then the client receives a NotFound error here that should prompt a return of `nil` instead of returning the error itself.

Reply:
yeah. I will add the additional check and see whether that resolves the issue.